shell/platform/darwin/macos/framework/Source/FLEEngine.mm

```diff
@@ -345,6 +345,38 @@ - (void)sendOnChannel:(nonnull NSString*)channel message:(nullable NSData*)messa
   }
 }
 
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  FlutterPlatformMessageResponseHandle* response_handle = nullptr;
+  struct Captures {
+    FlutterBinaryReply reply;
+  };
+  auto captures = new Captures();
+  captures->reply = callback;
+  auto message_reply = [](const uint8_t* data, size_t data_size, void* user_data) {
+    auto captures = reinterpret_cast<Captures*>(user_data);
+    NSData* reply_data = [NSData dataWithBytesNoCopy:(void*)data length:data_size freeWhenDone:NO];
+    captures->reply(reply_data);
+    delete captures;
+  };
+  FlutterPlatformMessageCreateResponseHandle(_engine, message_reply, captures, &response_handle);
```

Comment:
You need to check if the engine could create a response handle here. This call will return `kSuccess` in such cases. In case of failures, you will have to remember to delete the captures yourself because you aren't going to get the callback. I suggest using a unique pointer for the captures and only `release()` it on `kSuccess`.

Reply:
Done